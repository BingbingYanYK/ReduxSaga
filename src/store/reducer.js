import * as a from './actions';
import uniqBy from 'lodash/uniqBy';
import map from 'lodash/map';
import { combineReducers } from 'redux';

const initialPayment = {
  list: []
};

export function payment(state = initialPayment, action) {
  switch (action.type) {
    case a.FIND_PAYMENT.REQUEST:
      return {
        ...state,
        loading: true
      };
    case a.FIND_PAYMENT.SUCCESS:
      return {
        ...state,
        loading: false,
        list: uniqBy([...state.list, ...action.list], 'id'),
        found: map(action.list, 'id')
      };
    case a.FIND_PAYMENT.FAILURE:
      return {
        ...state,
        loading: false,
        found: []
      };
    default:
      return state;
  }
}

export function order(state = {}, action) {
  switch (action.type) {
    case a.SET_DESTINATION:
      return state; // TODO
    case a.SET_PAYMENT:
      return {
        ...state,
        payment: action.paymentId
      }
    case a.SUBMIT:
      return {
        ...state,
        departing: true
      }
    default:
      return state;
  }
}

export default combineReducers({
  payment,
  order
});
