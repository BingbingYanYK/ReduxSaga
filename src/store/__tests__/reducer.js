import { destination, payment, order } from '../reducer';

describe('store/reducer', () => {
  describe('#payment', () => {
    it('reduces a payment/FIND/REQUEST action', () => {
      const action = {
        type: 'payment/FIND/REQUEST',
        currencies: [0, 1]
      };
      const state = payment(undefined, action);
      expect(state).toMatchSnapshot();
    });

    it('reduces a payment/FIND/SUCCESS action', () => {
      const action = {
        type: 'payment/FIND/SUCCESS',
        list: [{ id: 0, name: 'test 1' }, { id: 1, name: 'test 2' }]
      };
      const state = payment(undefined, action);
      expect(state).toMatchSnapshot();
    });

    it('reduces a payment/FIND/FAILURE action', () => {
      const action = {
        type: 'payment/FIND/FAILURE',
        error: 'failure'
      };
      const state = payment(undefined, action);
      expect(state).toMatchSnapshot();
    });

    it('reduces an unknown action', () => {
      const action = {
        type: 'UNKNOWN'
      };
      const state = payment(undefined, action);
      expect(state).toMatchSnapshot();});
  });

  describe('#order', () => {
    it('reduces a destination/SET action', () => {
      const action = {
        type: 'destination/SET',
        destinationId: 0
      };
      const state = order(undefined, action);
      expect(state).toMatchSnapshot();
    });

    it('reduces a payment/SET action', () => {
      const action = {
        type: 'payment/SET',
        paymentId: 0
      };
      const state = order(undefined, action);
      expect(state).toMatchSnapshot();
    });

    it('reduces a order/SUBMIT action', () => {
      const action = {
        type: 'order/SUBMIT'
      };
      const state = order(undefined, action);
      expect(state).toMatchSnapshot();
    });

    it('reduces an unknown action', () => {
      const action = {
        type: 'UNKNOWN'
      };
      const state = order(undefined, action);
      expect(state).toMatchSnapshot();});
  });
});
