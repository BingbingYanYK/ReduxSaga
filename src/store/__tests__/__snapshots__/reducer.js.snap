// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store/reducer #order reduces a destination/SET action 1`] = `Object {}`;

exports[`store/reducer #order reduces a order/SUBMIT action 1`] = `
Object {
  "departing": true,
}
`;

exports[`store/reducer #order reduces a payment/SET action 1`] = `Object {}`;

exports[`store/reducer #order reduces an unknown action 1`] = `Object {}`;

exports[`store/reducer #payment reduces a payment/FIND/FAILURE action 1`] = `
Object {
  "found": Array [],
  "list": Array [],
  "loading": false,
}
`;

exports[`store/reducer #payment reduces a payment/FIND/REQUEST action 1`] = `
Object {
  "list": Array [],
  "loading": true,
}
`;

exports[`store/reducer #payment reduces a payment/FIND/SUCCESS action 1`] = `
Object {
  "found": Array [
    0,
    1,
  ],
  "list": Array [
    Object {
      "id": 0,
      "name": "test 1",
    },
    Object {
      "id": 1,
      "name": "test 2",
    },
  ],
  "loading": false,
}
`;

exports[`store/reducer #payment reduces an unknown action 1`] = `
Object {
  "list": Array [],
}
`;
